### PR TITLE
ON-38555 # allow `submissionService.getElementSubmissionValue` to acc…

### PR DESCRIPTION
**Pull request by divporter**
…ept an optional `elementId` property

### CODE REVIEW

File: src/replaceCustomValues.ts [hunk 1]
```diff
@@ -174,20 +175,69 @@ export function getElementSubmissionValue({
...
   formElements: FormTypes.FormElement[]
   submission: SubmissionTypes.S3SubmissionData['submission']
 } & ReplaceInjectablesFormatters): unknown {
-  const formElement = findFormElement(
-    formElements,
-    (element) =>
-      element.type !== 'page' &&
-      element.type !== 'section' &&
-      element.name === propertyName,
-  )
```
Comment (mymattcarroll) [inline src/replaceCustomValues.ts:187]: Don't we just need to change this to 

```
const formElement = findFormElement(
    formElements,
    (element) =>
      element.type !== 'page' &&
      element.type !== 'section' &&
      element.name === propertyName || element.id === elementId,
  )
```

Comment (divporter) [reply src/replaceCustomValues.ts:187]: Nup, fails the nested submission test

File: src/replaceCustomValues.ts [hunk 2]
```diff
@@ -174,19 +175,68 @@ export function getElementSubmissionValue({
...
+
+        if (Array.isArray(newSubmissionData) || !element.elements) {
+          continue
+        }
+
+        const result = getElementSubmissionValue({
+          elementId,
+          formElements: element.elements,
+          //@ts-expect-error let the recursion take care of it
+          submission: newSubmissionData,
```
Comment (Zaxist) [inline src/replaceCustomValues.ts:216]: Better to just cast this as `SubmissionTypes.S3SubmissionData['submission']` probably.

File: src/replaceCustomValues.ts [hunk 3]
```diff
@@ -174,19 +175,68 @@ export function getElementSubmissionValue({
...
+          formatDate,
+          formatDateTime,
+          formatTime,
+          formatNumber,
+          formatCurrency,
+        })
+        if (result) {
+          unknown = result
+          formElement = element
```
Comment (Zaxist) [inline src/replaceCustomValues.ts:225]: Recursive function needs to return element.

File: src/replaceCustomValues.ts [hunk 4]
```diff
@@ -174,19 +175,68 @@ export function getElementSubmissionValue({
...
+    }
+  } else if (propertyName) {
+    formElement = findFormElement(
+      formElements,
+      (element) =>
+        element.type !== 'page' &&
+        element.type !== 'section' &&
+        element.name === propertyName,
+    ) as FormTypes.FormElementWithName
+    unknown = submission[propertyName]
```
Comment (Zaxist) [inline src/replaceCustomValues.ts:237]: `propertyName` could be for a non-root element, right?

File: src/replaceCustomValues.ts [hunk 5]
```diff
@@ -215,6 +265,10 @@ export function getElementSubmissionValue({
       const value = unknown as string[]
       const selectedOptionLabels: string[] = value.reduce(
         (labels: string[], selectedOption: string) => {
+          //TS making me do this again
+          if (formElement?.type !== 'checkboxes') {
```
Comment (Zaxist) [inline src/replaceCustomValues.ts:271]: If you define `const options = ...` outside of the reduce you won't have to do this.

**Pull request merged**